src/main/java/com/xebialabs/deployit/ci/DeployitNotifier.java

```diff
@@ -308,6 +308,11 @@ public ListBoxModel doFillCredentialsIdItems(@AncestorInPath ItemGroup context)
         }
 
         public FormValidation doCheckCredential(@QueryParameter String credential, @AncestorInPath AbstractProject project) {
+            if (project == null) { // no context
+                return FormValidation.ok();
+            }
+            project.checkPermission(AbstractProject.CONFIGURE);
```

Comment:
Would be nicer to check `hasPermission` and return a useful value here, otherwise the UI will be weird for people with `Item.EXTENDED_READ` permission. This is documented on https://github.com/jenkinsci/credentials-plugin/blob/master/docs/consumer.adoc#providing-a-ui-form-element-to-let-a-user-select-credentials

Reply:
added hasPermission check

```
if (project == null) { // no context
    if (!project.hasPermission(Jenkins.ADMINISTER)) {
        return FormValidation.ok();
    }
} else if (!project.hasPermission(Item.EXTENDED_READ)
        && !project.hasPermission(CredentialsProvider.USE_ITEM)) {
    return FormValidation.ok();
}
```